068-performance-testing.md

```diff
@@ -0,0 +1,451 @@
+
+This approach may appear more effective than the initial one, yet it necessitates the creation of numerous support classes for:
+
+1. Extracting metrics from Kubernetes Pods components
+2. Monitoring the entire lifecycle
+3. Documenting (i.e., reporting) the outcomes of performance tests:
+   - this could all metrics scraped during performance test
+   - and then reporting via table of each parameter monitored during such test
+
+Consider eliminating points 2 and 3 by utilizing an existing performance tool that encompasses all these functionalities.
+
+### Approach 3: Using known performance tools
+
+[Grafana k6](https://k6.io/) is an open-source performance testing tool designed to assess the load capacity and performance of various systems and infrastructure.
+It allows developers and testers to simulate traffic and user behavior towards applications, APIs,
+and services to identify potential bottlenecks and performance limits under controlled test environments.
+k6 integrates seamlessly with [Grafana](https://grafana.com/), a popular open-source analytics and monitoring solution,
+enabling users to visualize test results and performance metrics in real-time through comprehensive dashboards.
+This integration facilitates easy analysis and interpretation of data, making it simpler for teams to make informed
+decisions based on the performance characteristics of their applications.
+
+One of the key features of k6 is its extensibility through extensions, which allows users to tailor the tool to their specific testing needs.
+For instance, the ability to extend k6's functionality to work with specific technologies like Kubernetes through
+the xk6-kubernetes extension showcases its flexibility.
+Similarly, this extensibility opens up the possibility of developing a dedicated extension for Strimzi, enabling specialized
+performance testing for Strimzi-managed Apache Kafka clusters.
+This approach would allow for targeted load testing of Kafka operators, providing insights into how Strimzi components
+behave under various load conditions, thereby ensuring that Strimzi deployments are optimized for performance and reliability.
+
+The primary challenge is that developing such an extension requires proficiency in Go/Javascript,
+which might be less familiar to our community accustomed to Java.
+
+####  Notes on Release Cycle for Using k6:
+
+The release cycle for incorporating k6 into Strimzi testing would bear similarities to the dedicated repository approach,
+with both pros and cons.
+One advantage is the potential for rapid adoption and integration of new performance testing methodologies,
+given k6's established user base and continuous updates.
+However, a possible downside is the need to maintain compatibility with Strimzi's evolving infrastructure,
+possibly requiring frequent updates to any Strimzi-specific extensions or configurations within k6.
+
+## Selected Approach/Proposal: Approach 1 - Integration within the strimzi-kafka-operator
```

Comment:
I would keep just this selected approach and others leave to rejected alternatives
In case that others will agree on different approach, you will have it as one of the commits, so you can just easily replace it 🤷

Reply:
Removed